README

				            mPopup by Prashant Kumar
			             	    ========================
			             	      © version: 1.0.0.2
				            	  3-Apr-2014
					             2:55PM
Mail : kr.prashant94@gmail.com
Facebook : https://facebook.com/prashant94
--------------------------------------------------------------------
					             INTRODUCTION
mPopup is a API plug-in that offers a nice and elegant way to alert, 
zooming, prompt and other popup functionality for images, html and 
other web content on your webpages.

More information and examples : http://pkrcciit.blogspot.in

				
--------------------------------------------------------------------

To add the mPopup you just include this code in your page <head> tag : -

	<HEAD>
		<script type="text/javascript" src="http://ajax.googleapis.com/ajax/libs/jquery/1.7/jquery.min.js"></script>
		<link href="/mPopup/mPopup.css" rel="stylesheet" type="text/css" />
		<script type="text/javascript" src="/mPopup/mPopup.js"></script>
	</HEAD>

Note : - 	1. To trigger the mPopup it must be include jQuery file

------------------ Triggring mPopup --------------------------------

Use the following script to trigger the mPopup :-
			    $('*').mPopup({"options1", "options2"...});
Eg :- 
			$('*').mPopup({
				heading	: "My heading",
				url		: "test.html",
				footer	:'My footer',
				progress: '#149bdf',
				type	: ""
			});
			
--------------------------------------------------------------------
						OPTIONS
A.	AJAX
		1.	method 		: GET/POST
		2.	send		: Data send
		3.	url			: path of the file.
B.	Text Display
		1.	heading		: Text on heading
		2.	footer		: Text on footer
C.	Style
		1.	progress	: Progress bar color (is not set Progress false )
		2.	type		: file (only)
		3.	Color_foot	: Footer text color
		4.	color_back	: Footer background color
		5.	top			: Possition to the top
		6.	width		: Width of the popup
		7.	height		:  Height of the popup
		8.	padding		: Padding on popup
--------------------------------------------------------------------
			Have a bug(Error)

Please create an issue or mail me 


CHANGE_LOG
--------------------------------------------------------------------
--------------------------------------------------------------------
Version: 1.0.0.2 (4-Apr-2014, 2:55PM)
Features Added
	2. fadeIn(200) and fadeOut(200) are use for appearing and exit on popup.
Features Upgread
	1. Default footer style changed.
Features Removed
	Not remove any features.
Bug Fixed
	1. If heading is not set remove head "<div>".
	2. Fix the position of the close button
--------------------------------------------------------------------
--------------------------------------------------------------------
Version: 1.0.0.1 (3-Apr-2014, 9:35PM)
Relished the first mPopup API using jQuery JavaScript and CSS, CSS3
Features
	1. Popup using AJAX request.
	2. Popup opens another web page "type = file" only valid format.
	3. GET and POST method support.
	4. Data sending to another url support.
Changeable Property
	A.	AJAX
		1.	method 		: GET/POST
		2.	send		: Data send
		3.	url			: path of the file.
	B.	Text Display
		1.	heading		: Text on heading
		2.	footer		: Text on footer
	C.	Style
		1.	trogress	: Progress bar color (is not set Progress false )
		2.	type		: file (only)
		3.	Color_foot	: Footer text color
		4.	color_back	: Footer background color
		5.	top			: Possition to the top
		6.	width		: Width of the popup
		7.	height		:  Height of the popup
		8.	padding		: Padding on popup

mPopup.js
(function($) {
	p = $.fn.mPopup = function(e) {
		var data = jQuery.extend({}, jQuery.fn.mPopup, e);
		var center = (window.innerWidth/2)-(data.width/2);
		$('body').append('<div id="mPopup"> <div class="content"> <span class="img" onClick="mPopup_close();"></span> <div class="head"></div> <div class="body"></div><br/><div class="foot"></div> </div> </div>');
		$('#mPopup').hide();
		$('body').css('overflow','hidden');
		$('#mPopup .content').addClass('morus_content');
		$('#mPopup .body').css({'height':data.height+'px'});		
		$('#mPopup .morus_content').css({'top':data.top+'px', 'left':center, 'width':data.width+'px', 'height':data.height+'px'});
		$('#mPopup').fadeIn(200);
		if(data.heading != null){
			$('#mPopup .body').css({'height':data.height-30+'px'});
			$('#mPopup .head').addClass('morus_head');
			$('#mPopup .head').html(data.heading);
		}else{$('#mPopup .head').remove();}
		if(data.url.length > 3 && data.type == 'file'){
			sender(data.method, data.send, data.url);
		}else if(data.url.length > 3 && data.type == 'image'){
			$('#mPopup .body').html('<img src="'+data.url+'" style="width:'+data.width+'px;height:'+data.height+'px;" >');
			$('#mPopup .head').remove();
		}
		if(data.footer != null){
			if(data.progress == 0){
				$('#mPopup .foot').css({'text-align':'center'});
				$('#mPopup .foot').html('<span class="morus_foot">'+data.footer+'</span>');
				$('#mPopup .morus_foot').addClass('norm');
				$('#mPopup .morus_foot').css({'background':data.back_foot, 'color':data.color_foot});
			}else{
				$('#mPopup .foot').addClass('morus_foot');
				$('#mPopup .morus_foot').html(data.footer);
				$('#mPopup .morus_foot').addClass('morus_progress');
				$('#mPopup .morus_progress').css({'background-color':data.progress, 'color':data.color_foot});
			}
		}
		$(window).resize(function(){
			$('#mPopup .content').css('left', (window.innerWidth/2)-(data.width)/2);
		});
	};
	$.extend(p, {
		version		: '1.0.0.2',
		/*-----  AJAX request  -------*/
		method		: 'POST',
		send		: null,
		url			:null,
		/*---- mPopup Styles   -----*/
		heading		: null,
		footer		: null,
		progress	: 0,
		type		: 'file',
		/*----- Content Style ------*/
		color_foot	: '#FFF',
		back_foot	: 'rgba(0, 0, 0, 0.8)',
		top			: '150',
		width		: '500',
		height		: '200',
		padding		: '10'
		
	});
	function sender(type, data, url){
		$.ajax({
			type	: type,
			data	: data,		
			url		: url,
			timeOut	: 5000,
			dataType: 'html',
			success	: function(html, textStatus) {
				$('#mPopup .body').html(html);
			},
			error: function(xhr, textStatus, errorThrown) {
				$('#mPopup .body').html("Unable to Connect...");
			}
		});
	}
})(jQuery);
function mPopup_close(){
	$('#mPopup').fadeOut(200,function(){$('#mPopup').remove();});
	$('body').css('overflow','auto');
}

$(window).bind('keydown', function(e){
	if(e.keyCode === 27){
		mPopup_close();
	}
});

js/index.php
<?php header("Content-type: text/javascript"); ?>
(function($) {
	p = $.fn.mPopup = function(e) {
		var data = jQuery.extend({}, jQuery.fn.mPopup, e);
		var center = (window.innerWidth/2)-(data.width/2);
		$('body').append('<div id="mPopup"> <div class="content"> <span class="img" onClick="mPopup_close();"></span> <div class="head"></div> <div class="body"></div><div class="foot"></div> </div> </div>');
		$('#mPopup').hide();
		$('body').css('overflow','hidden');
		$('#mPopup .content').addClass('morus_content');
		$('#mPopup .body').css({'height':data.height+'px'});		
		$('#mPopup .morus_content').css({'top':data.top+'px', 'left':center, 'width':data.width+'px', 'height':data.height+'px'});
		$('#mPopup').fadeIn(200);
		if(data.heading != null){
			$('#mPopup .body').css({'height':data.height-30+'px'});
			$('#mPopup .head').addClass('morus_head');
			$('#mPopup .head').html(data.heading);
		}else{$('#mPopup .head').remove();}
		if(data.url.length > 3 && data.type == 'file'){
			sender(data.method, data.send, data.url);
		}else if(data.url.length > 3 && data.type == 'image'){
			$('#mPopup .body').html('<img src="'+data.url+'" style="width:'+data.width+'px;height:'+data.height+'px;" >');
			$('#mPopup .head').remove();
		}
		if(data.footer != null){
			if(data.progress == 0){
				$('#mPopup .foot').css({'text-align':'center'});
				$('#mPopup .foot').html('<span class="morus_foot">'+data.footer+'</span>');
				$('#mPopup .morus_foot').addClass('norm');
				$('#mPopup .morus_foot').css({'background':data.back_foot, 'color':data.color_foot});
			}else{
				$('#mPopup .foot').addClass('morus_foot');
				$('#mPopup .morus_foot').html(data.footer);
				$('#mPopup .morus_foot').addClass('morus_progress');
				$('#mPopup .morus_progress').css({'background-color':data.progress, 'color':data.color_foot});
			}
		}
		$(window).resize(function(){
			$('#mPopup .content').css('left', (window.innerWidth/2)-(data.width)/2);
		});
	};
	$.extend(p, {
		version		: '1.0.0.2',
		/*-----  AJAX request  -------*/
		method		: 'POST',
		send		: null,
		url			:null,
		/*---- mPopup Styles   -----*/
		heading		: null,
		footer		: null,
		progress	: 0,
		type		: 'file',
		/*----- Content Style ------*/
		color_foot	: '#FFF',
		back_foot	: 'rgba(0, 0, 0, 0.8)',
		top			: '150',
		width		: '500',
		height		: '200',
		padding		: '10'
		
	});
	function sender(type, data, url){
		$.ajax({
			type	: type,
			data	: data,		
			url		: url,
			timeOut	: 5000,
			dataType: 'html',
			success	: function(html, textStatus) {
				$('#mPopup .body').html(html);
			},
			error: function(xhr, textStatus, errorThrown) {
				$('#mPopup .body').html("Unable to Connect...");
			}
		});
	}
})(jQuery);
function mPopup_close(){
	$('#mPopup').fadeOut(200,function(){$('#mPopup').remove();});
	$('body').css('overflow','auto');
}

$(window).bind('keydown', function(e){
	if(e.keyCode === 27){
		mPopup_close();
	}
});